uefi/src/proto/mod.rs

```diff
@@ -81,3 +81,57 @@ pub mod shell_params;
 pub mod shim;
 pub mod string;
 pub mod tcg;
+
+/// The UEFI boot policy is a property that influences the behaviour of
+/// various UEFI functions that load files (typically UEFI images).
+///
+/// This type is not ABI compatible. On the ABI level, this is an UEFI
+/// boolean.
+#[derive(Debug, Copy, Clone, PartialOrd, PartialEq, Eq, Ord)]
+pub enum BootPolicy {
+    /// Indicates that the request originates from the boot manager, and that
+    /// the boot manager is attempting to load the provided `file_path` as a
+    /// boot selection.
+    ///
+    /// Boot selection refers to what a user has chosen in the (GUI) boot menu.
+    TRUE,
+    /// The provided `file_path` must match an exact file to be loaded.
+    FALSE,
+}
+
+impl From<BootPolicy> for bool {
+    fn from(value: BootPolicy) -> Self {
+        match value {
+            BootPolicy::TRUE => true,
+            BootPolicy::FALSE => false,
+        }
+    }
+}
+
+impl From<bool> for BootPolicy {
+    fn from(value: bool) -> Self {
+        match value {
+            true => Self::TRUE,
+            false => Self::FALSE,
+        }
+    }
+}
+
+impl From<BootPolicy> for u8 {
+    fn from(value: BootPolicy) -> Self {
+        match value {
+            BootPolicy::TRUE => 1,
+            BootPolicy::FALSE => 0,
+        }
+    }
+}
+
+impl From<u8> for BootPolicy {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => Self::FALSE,
+            1 => Self::TRUE,
+            _ => panic!("Invalid variant"),
```

Comment:
Per https://doc.rust-lang.org/std/convert/trait.From.html, `From` must not be used if it can fail; should use `TryFrom` instead.

Reply:
fixed